tests/Unit/LegacyConverters/DomainToLegacyConverterTest.php

```diff
@@ -5,159 +5,160 @@
+		$this->assertEquals( $paypalData['paypal_subscr_id'], $paypalInfo['paypal_subscr_id'] );
+		$this->assertEquals( $paypalData['paypal_payer_status'], $paypalInfo['paypal_payer_status'] );
+		$this->assertEquals( $paypalData['paypal_mc_gross'], $paypalInfo['paypal_mc_gross'] );
+		$this->assertEquals( $paypalData['paypal_mc_currency'], $paypalInfo['paypal_mc_currency'] );
+		$this->assertEquals( $paypalData['paypal_mc_fee'], $paypalInfo['paypal_mc_fee'] );
+		$this->assertEquals( $paypalData['paypal_settle_amount'], $paypalInfo['paypal_settle_amount'] );
+		$this->assertEquals( $paypalData['ext_payment_id'], $paypalInfo['ext_payment_id'] );
+		$this->assertEquals( $paypalData['ext_payment_status'], $paypalInfo['ext_payment_status'] );
+		$this->assertEquals( $paypalData['ext_payment_timestamp'], $paypalInfo['ext_payment_timestamp'] );
 	}
 
-	public function testWhenGivenDirectDebitDoctrineApplication_setsDirectDebitFieldsInMembershipApplication() {
-		$this->markTestIncomplete( 'This will work again when we update the legacy converters' );
+	public function testWhenGivenDirectDebitPayment_setsDirectDebitFieldsInMembershipApplication() {
 		$doctrineApplication = new DoctrineApplication();
 		$application = ValidMembershipApplication::newDomainEntity();
 
+		$paymentData = new LegacyPaymentData( 1, 1, 'UEB', [
+			'konto' => 'account',
+			'blz' => 'bank code',
+			'bankname' => 'bank name',
+			'bic' => 'biccybic',
+			'iban' => 'ibannabi',
+		], LegacyPaymentStatus::EXTERNAL_BOOKED->value );
+
 		$converter = new DomainToLegacyConverter();
-		$converter->convert( $doctrineApplication, $application );
+		$converter->convert( $doctrineApplication, $application, $paymentData );
 
-		/** @var \WMDE\Fundraising\PaymentContext\Domain\Model\DirectDebitPayment $payment */
-		$payment = $application->getPayment()->getPaymentMethod();
-		$bankData = $payment->getBankData();
+		$bankData = $paymentData->paymentSpecificValues;
 
-		$this->assertEquals( $bankData->getAccount(), $doctrineApplication->getPaymentBankAccount() );
-		$this->assertEquals( $bankData->getBankCode(), $doctrineApplication->getPaymentBankCode() );
-		$this->assertEquals( $bankData->getBankName(), $doctrineApplication->getPaymentBankName() );
-		$this->assertEquals( $bankData->getBic(), $doctrineApplication->getPaymentBic() );
-		$this->assertEquals( $bankData->getIban(), new Iban( $doctrineApplication->getPaymentIban() ) );
+		$this->assertEquals( $bankData['konto'], $doctrineApplication->getPaymentBankAccount() );
+		$this->assertEquals( $bankData['blz'], $doctrineApplication->getPaymentBankCode() );
+		$this->assertEquals( $bankData['bankname'], $doctrineApplication->getPaymentBankName() );
+		$this->assertEquals( $bankData['bic'], $doctrineApplication->getPaymentBic() );
+		$this->assertEquals( $bankData['iban'], $doctrineApplication->getPaymentIban() );
 	}
 
 	public function testWhenGivenNonModeratedNonCancelledApplicationWithUncompletedExternalPayment_setsNeutralStatus() {
```

Comment:
Probably don't need this test, it may have been originally added as a safeguard or something.